vendor/bat-native-ads/src/bat/ads/internal/features/features.cc

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/features/features.h"
+
+#include "base/metrics/field_trial.h"
+#include "base/metrics/field_trial_params.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+
+namespace ads {
+namespace features {
+
+namespace {
+const char kPageProbabilitiesStudyName[] = "PageProbabilitiesHistoryStudy";
+const int kDefaultHistorySize = 5;
+}  // namespace
+
+// Controls behaviour of the contextual ad matching mechanism, e.g. by
+// adjusting the number of page classifications used to infer user interest.
+const base::Feature kContextualAdsControl { "ContextualAdsControl",
+    base::FEATURE_DISABLED_BY_DEFAULT };
+
+bool IsPageProbabilitiesStudyActive() {
+  if (!base::FieldTrialList::Find(kPageProbabilitiesStudyName)) {
+    return false;
+  }
+
+  return true;
+}
+
+std::string GetPageProbabilitiesStudy() {
+  return kPageProbabilitiesStudyName;
+}
+
+std::string GetPageProbabilitiesFieldTrialGroup() {
+  std::string group_name;
+  base::FieldTrial* field_trial = base::FieldTrialList::Find(
+      kPageProbabilitiesStudyName);
+  if (!field_trial) {
+    return group_name;
+  }
+
+  group_name = field_trial->group_name();
+  return group_name;
+}
+
+int GetPageProbabilitiesHistorySize() {
+  return GetFieldTrialParamByFeatureAsInt(
+      kContextualAdsControl,
+      "page_probabilities_history_size",
+      kDefaultHistorySize);
+}
+
+std::string GetPageProbabilitiesStudyLog() {
```

Comment:
nit: this seems a bit odd to me, I think it makes more sense to have this method just log what you want instead of returning a string to use inside a log command somewhere else. So `void LogPageProbabilitiesStudyStatus()` or something

Reply:
changed!